analyses/01_sample_distribution_analysis/01-tumor-descriptor-and-assay-count-plot.Rmd

```diff
@@ -0,0 +1,285 @@
+```
+
+# Set up
+
+```{r load-library}
+
+suppressPackageStartupMessages({
+  library(tidyverse)
+  library(ggplot2)
+  library(flextable)
+  library(RColorBrewer)
+})
+
+
+```
+
+## Directories and File Inputs/Outputs
+
+```{r set-dir-and-file-names}
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper sourcing of functions no
+# matter where this is called from
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+setwd(root_dir)
+
+
+# File path to input directory
+input_dir <-
+  file.path(root_dir, "input")
+
+
+# Inputs
+input_metadata <- file.path(input_dir, "histologies.tsv")
+input_cols <- file.path(input_dir, "PBTA-germline-histology-groups-plot-mapping.tsv")
+
+
+# File path to results directory
+results_dir <-
+  file.path(root_dir, "results")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+
+# File path to plots directory
+plots_dir <-
+  file.path(root_dir, "plots")
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+
+# Outputs
+output_PBTA_filter <- file.path(results_dir, "data_by_cohort_PBTA_filter.tsv")
+output_PBTA_filter_count <- file.path(results_dir, "data_by_cohort_PBTA_filter_count.tsv")
+```
+
+
+## Load data
+```{r load-inputs-please-wait}
+# Inputs
+## Read in metadata
+histologies <- readr::read_tsv(input_metadata, guess_max = 100000, show_col_types = FALSE)
+# dim(data)
+
+# Read table with colors per histology
+cols <- readr::read_tsv(input_cols, guess_max = 100000, show_col_types = FALSE) # Load plotting colors
+
+# Add colors palette to the histologies file
+hist_cols <- histologies %>% left_join(cols, by=c('cancer_group','broad_histology'))  #to merge tables
+# This is the tidyverse method. Try not to name variables the same as a function's name - that may cause coding errors downstream.
+
+
+# to discard when add the col palette for PBTA cohort
+mypalette <- brewer.pal(12,"Paired") # colorblindFriendly = TRUE
+col5 <- colorRampPalette(mypalette)(5)
+col9 <- colorRampPalette(mypalette)(9)
+
+```
+
+# Process Data
+
+All tumor samples have matched normal samples. 
+We need to filter normal samples out for further analysis. 
+We are using "pathology_diagnosis" instead of "sample_type". This way we discard normal samples (as shown in "sample_type") in addition to undiagnosed tumor samples.
+Also, we will discard Derived Cell Line, PDX from "composition".
+
+
+```{r cohort PBTA, echo=TRUE}
+data <- hist_cols  %>%
+   filter(cohort == "PBTA",
+          !is.na(pathology_diagnosis),
+          !composition %in% c("Derived Cell Line", "PDX")) %>% 
+         # !is.na(cancer_group) # Discard NAs in cancer type
+  mutate(tumor_descriptor = ifelse(sample_id == "C4106355-Tumor" & tumor_descriptor == "Not Reported", "Recurrence", 
+                                   ifelse(sample_id == "C4113612-Tumor" & tumor_descriptor == "Initial CNS Tumor", "Recurrence", 
+                                          tumor_descriptor)))
```

Comment:
```suggestion
  mutate(tumor_descriptor = ifelse(sample_id == "C4106355-Tumor", "Recurrence", tumor_descriptor,
                                   ifelse(sample_id == "C4113612-Tumor", "Initial CNS Tumor", tumor_descriptor)))
```

The second ifelse should be if "Not Reported" then "Initial CNS Tumor", but I think it is easier to just remove the if not reported part bc we know what they should be regardless..